src/BloomBrowserUI/bookEdit/js/synphonyApi.js

```diff
@@ -44,78 +38,64 @@ var SynphonyApi = (function () {
             }
         }
     };
-
     SynphonyApi.prototype.loadFromLangData = function (langData) {
         if (!this.source)
             this.source = new ReaderSettings();
-
         if (this.source.letters === '') {
             var sorted = langData.LanguageSortOrder.join(' ').toLowerCase().split(' ');
             sorted = _.uniq(sorted);
-
             this.source.letters = sorted.join(' ');
         }
     };
-
     SynphonyApi.fireCSharpEvent = function (eventName, eventData) {
-        //noinspection TaskProblemsInspection
-        var event = new MessageEvent(eventName, { 'view': window, 'bubbles': true, 'cancelable': true, 'data': eventData });
-        document.dispatchEvent(event);
+        fireCSharpAccordionEvent(eventName, eventData);
```

Comment:
Calling a  function in UI code, from a function in model code, that talks to more UI code somewhere... this is smelling really bad.